### PR TITLE
chungus cakc refactor

### PR DESCRIPTION
**Pull request by crewtool**
# Description
This PR refactors `chungus` in the way that model does not have references to the same object in its descendants (it is needed by FSDP).
Also restored `chungus` test.

# Checklist
   - [x] Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - [x] Are all variables set with logical defaults that are backward compatible?
   - [x] I attached link to neptune if it was necessary
   - [x] Have you successfully executed the linter and tests?
   - [x] Are all functions concise and don't require splitting?
   - [x] Is there documentation for arguments and complex logic?
   - [x] Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - [x] Are all functions placed in appropriate files?
   - [x] Is the PR name meaningful and descriptive?
   - [x] Is PR description provided, or unnecessary?
  
Other questions:
   - [x] If I made changes to the requirements or anything related to the Singularity image I did generate new image and upload a new image to the clusters
   - [x] If this change is significant enough I notified all individuals working with this repository (@channel in slack channel)
   - [x] If there is a need for a second reviewer I requested additional review 

# Reviewer's checklist:
   - [ ] Does the Neptune link work and does it actually compare the situation before and after the PR?
   - [ ] Do I understand the code?
   - [ ] Do I think another reviewer is not needed or I commented that this PR need another reviewer?

### CODE REVIEW

File: lizrd/core/llm.py [hunk 1]
```diff
@@ -433,7 +425,6 @@ def __init__(self, embedding_layer, encoder_tower, head):
                 ]
             )
         )
-        self.head = head
```
Comment (kuba-krj) [inline lizrd/core/llm.py:436]: Maybe we can just have `self.embedding_layer`, `self.encoder_tower`, `self.head`  instead of `self.full_model`?

**Review by kuba-krj — APPROVED** [review]
LGTM. Just one suggestion about design

File: lizrd/core/llm.py (hunk 1 above)
Comment (crewtool) [reply lizrd/core/llm.py:436]: Then we also need to change forward

File: lizrd/core/test_misc.py [hunk 2]
```diff
@@ -9,9 +9,13 @@
     EinMix,
 )
 from lizrd.core.misc import Chungus
-from lizrd.datasets.wikibookdata import get_processed_dataset
-from lizrd.support.test_utils import GeneralTestCase, heavy_test, skip_test
+from research.datasets import get_processed_dataset
```
[the remark this answers is not in the document]
Comment (crewtool) [reply lizrd/core/test_misc.py:12]: @maciek-pioro moved get_processed_dataset

File: lizrd/core/llm.py (hunk 1 above)
Comment (crewtool) [reply lizrd/core/llm.py:436]: But actually that actually that should look better than this

File: research/conditional/utils/model_utils.py [hunk 3]
```diff
@@ -71,7 +71,7 @@ def chungized_llm_loss(
 
     def make_custom_forward():
         def custom_forward(*inputs):
-            output = model.head(inputs[0])
+            output = model.full_model.head(inputs[0])
             with torch.autocast(device_type="cuda", enabled=False, dtype=torch.float16):
                 gt = inputs[1]
                 mask = inputs[2]
```
Comment (Simontwice) [inline research/conditional/utils/model_utils.py:77]: this is major ugly, could we have sth like `x, gt, mask = inputs`? If not in this PR, maybe an issue with this?

Comment (kuba-krj) [reply research/conditional/utils/model_utils.py:77]: Orthogonal

File: lizrd/core/llm.py (hunk 1 above)
Comment (maciek-pioro) [reply lizrd/core/llm.py:436]: It's unusual to create an `nn.Module` on each forward. I don't know if something might break because of that, but kinda looks like that. Let's resolve this conversation if there is we have no other idea how to do what we do here.

**Review by maciek-pioro — APPROVED** [review]
Approved but look at my comment and think about it for a second.

File: research/conditional/utils/model_utils.py [hunk 4]
```diff
@@ -71,7 +71,8 @@ def chungized_llm_loss(
 
     def make_custom_forward():
         def custom_forward(*inputs):
-            output = model.head(inputs[0])
+            x, gt, mask = inputs
+            output = model.head(x)
             with torch.autocast(device_type="cuda", enabled=False, dtype=torch.float16):
                 gt = inputs[1]
                 mask = inputs[2]
```
Comment (Simontwice) [inline research/conditional/utils/model_utils.py:78]: we have this already, @crewtool
